### ISSUE TITLE
QRcode not working anymore (change of url in a service xerte uses)

### ISSUE THREAD

**Issue by torinfo** [thexerteproject/xerteonlinetoolkits#509]
The service the QRcode page uses to generate a short url has been permanently moved to https

The 301 error is displayed above the QRcode. 

Fix is simple: change the http on line 37 to https

**Comment by ronm123** [comment]
Hi Tom
I've just been testing this and sure enough existing LO's with a QR Code page included are now showing an error in place of the short url 301 moved permanently e.g.
![image](https://cloud.githubusercontent.com/assets/5167146/15470371/3cd0735c-20e8-11e6-9126-2fc88b496cac.png)

However although replacing the QR code page model removes this error it doesn't seem to show the short url. Did your fix just remove the error or should it show the short url too?

Is this working for others?

**Comment by torinfo** [comment]
I really only added an 's to https. I didn't touch anything else.

Ron Mitchell <notifications@github.com> schreef op 23 mei 2016 08:26:46 GMT-04:00:
>Hi Tom
>I've just been testing this and sure enough existing LO's with a QR
>Code page included are now showing an error in place of the short url
>301 moved permanently e.g.
>![image](https://cloud.githubusercontent.com/assets/5167146/15470371/3cd0735c-20e8-11e6-9126-2fc88b496cac.png)
>
>However although replacing the QR code page model removes this error it
>doesn't seem to show the short url. Did your fix just remove the error
>or should it show the short url too?
>
>Is this working for others?
>
>---
>You are receiving this because you modified the open/close state.
>Reply to this email directly or view it on GitHub:
>https://github.com/thexerteproject/xerteonlinetoolkits/issues/509#issuecomment-220965985

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by ronm123** [comment]
Looks like they may have changed the way their api works. https://is.gd/apishorteningreference.php 

I've tried changing the code to `shorturl="rss_proxy.php?rss=https://is.gd/create.php?format=simple&url=" + pastedurl;` but that isn't displaying the url either. 

I notice there is also a json format. Might have always been but wondering if we can somehow use that and bypass the need for rssproxy too?

**Comment by JohnSmith-LT** [comment]
The problem seems to be with Snoopy in rss_proxy, if you try this url

`https://is.gd/api.php?longurl=https://api.jquery.com/jquery.ajax/`

Then you get back the correct short url but if you try

`rss_proxy.php?rss=https://is.gd/api.php?longurl=https%3A%2F%2Fapi.jquery.com/jquery.ajax%2F`

on your install then you should get the same, but I get nothing displayed??

Also, we can't use the JSON interface as it violates the Same Origin Policy when in offline mode or hosted locally (xampp) and you get 

`No 'Access-Control-Allow-Origin' header is present on the requested resource.`

I remember contacting them to see if they would allow it and they refused since there would be no way to block an individual url for breaching their rate limiting.

Also, we may be in breach of their terms as they say at the bottom of that page on their website that best practice should be to show the short url as the result of a button press and not as a result of page load as it increases their bandwidth...

**Comment by JohnSmith-LT** [comment]
I've upgraded Snoopy and it seems to have resolved the issue, for me anyway...

**Comment by ronm123** [comment]
Hi John I've pulled down the latest develop which obviously includes your changes and I don't get the error display now but I don't get the short url either. Do you see the short url displayed above the QR Code?

Also re their terms and conditions in a way we do only show the page and use the code on button click e.g. the next button which navigates to the page. On the other hand I guess we could add a button to the page e.g. on page load display the full url then on click of the button replace that with the shortened version.

**Comment by JohnSmith-LT** [comment]
Hi Ron, good point about the button click and I don't suppose there will be high enough traffic from a single instance that would cause them too much issue...

I just tried the QRCode again and it's working fine for me although i'm on Xampp so it could be something else at play I suppose:

![image](https://cloud.githubusercontent.com/assets/3907362/15536029/a2c2d288-2267-11e6-9754-4730fda02207.png)

**Comment by ronm123** [comment]
Hi John
I was getting that error on xampp and on a live install. Now that Tom fixed the 'no pages' issue I'll update the live install with the same code and report back.

**Comment by ronm123** [comment]
Hi John
I updated an online install and that is showing the short url ok now. My xampp install with the same code isn't showing the url. Cleared offline files just in case but not difference. Not sure why xampp install isn't showing the shorturl but online install is, especially when the same xampp install has worked previously, but not a major problem if it works online which it seems to. Will close the issue and let someone else re-open if they have similar problems.

**Comment by JohnSmith-LT** [comment]
Hmm don't really know why that would be, i've updated the picture above which shows that mine (on Xampp) works just fine...

**Comment by ronm123** [comment]
Hi John no idea either. Have tried the same xampp install on a different laptop same problem.
Tried using my 3g signal for internet connection instead of network in case block at their end due to requests from my ip/isp but no difference. Not seeing any errors in browser console just not seeing the short url in the page. But it works online so doesn't really matter. Would be curious if others can reproduce though?

**Comment by JohnSmith-LT** [comment]
Yes quite strange. Have you tried putting the urls that you see in network (the one containing rss_proxy) into the browser? Then try removing rss_proxy.php part. So in the example above:

`https://is.gd/api.php?longurl=https://api.jquery.com/jquery.ajax/`

and

`YOUR_PATH_TO_XOT/rss_proxy.php?rss=https://is.gd/api.php?longurl=https%3A%2F%2Fapi.jquery.com/jquery.ajax%2F`

both should give short link.

Might help narrow things down.

**Comment by ronm123** [comment]
Hi John
I had tested with the url I have in the LO for testing which is just http://nottingham.ac.uk/xerte and using https://is.gd/api.php?longurl=http%3A%2F%2Fnottingham.ac.uk%2Fxerte&_=1464185985946 in the address bar works and shows the shorturl but I hadn't tested with the rss_proxy link direct. The rss_proxy link direct in the address bard doesn't display the short url:
http://127.0.0.1/develop/rss_proxy.php?rss=https://is.gd/api.php?longurl=http%3A%2F%2Fnottingham.ac.uk%2Fxerte&_=1464185985946

**Comment by JohnSmith-LT** [comment]
It must be Snoop that's causing it... Mine wasn't working at all and started working when I went to the latest version but perhaps if you have time you could try the previous versions to see if they work for you and we can try to locate what is breaking it, in Windows at least

https://sourceforge.net/projects/snoopy/files/Snoopy/

**Comment by ronm123** [comment]
Hi John just tried a few different copies/versions of Snoopy.class.php from both the snoopy downloads and previous Xerte installs none making a difference.

Thinking about your reference to Windows though could there be a case sensitivity issue somewhere explaining why it works on the linux installations but not local xampp? That said I would have though that would show up in the console.

**Comment by JohnSmith-LT** [comment]
I'll take a look tomorrow and try to trace the requests in Fiddler. I'd have thought it would have shown up before if it was a Windows/Linux thing though.

**Comment by JohnSmith-LT** [comment]
Sorry Ron, installed XAMPP on Windows, downloaded and installed xot v3.2 from community site
Tried QRCode page and got no short url, then I upgraded snoopy to the version from Github/develop and tried page again and it works fine... have tried tracing the issue with Fiddler in both cases and requests and responses are identical (except for no data contained in the response using the old snoopy)...

I'm at a loss. Can anyone else ( @FayCross or @torinfo ) confirm seeing this issue with the QRCode.html and snoopy file from the develop branch (with latest updates)? To recap - latest files on both xampp installs (mac/windows) on my computer work fine with the latest files and on Ron's the short url doesn't load and in fact running rss_proxy.php in the browser (full url) gives no response.
